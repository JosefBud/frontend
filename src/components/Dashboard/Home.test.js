import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import ConnectedHome, { DashboardHome } from './Home';
import { withMemoryRouter } from 'utils';

describe('Dashboard <Home />', () => {
  const stats = {
    activatedUsers: 1991,
    flavorTags: 90,
    flavors: 8617,
    recipeTags: 55,
    recipes: 42,
    tags: 115,
    userTokens: 300,
    users: 2000,
    vendors: 98
  };
  const mockStore = configureStore();
  const store = mockStore({
    dashboard: { stats }
  });
  const actions = {
    requestStats: jest.fn()
  };
  const RoutedHome = withMemoryRouter(ConnectedHome);

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RoutedHome />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('can refresh', () => {
    const component = renderer.create(
      <Provider store={store}>
        <DashboardHome actions={actions} stats={stats} />
      </Provider>
    );
    const { instance } = component.root.findByType(DashboardHome);

    expect(instance).toBeDefined();
    instance.refresh();
    expect(actions.requestStats).toHaveBeenCalledWith();
  });
});
