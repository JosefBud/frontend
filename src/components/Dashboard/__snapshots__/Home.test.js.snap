// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard <Home /> renders correctly 1`] = `
Array [
  <div
    className="card-columns"
  >
    <div
      className="card bg-danger text-white"
    >
      <div
        className="card-header"
      >
        Users: 
        2000
      </div>
      <div
        className="card-body"
      >
        <h3>
          Activated Users: 
          1991
        </h3>
        <h3>
          Inactive Users: 
          9
        </h3>
        <h3>
          Active Tokens: 
          300
        </h3>
      </div>
    </div>
    <div
      className="card bg-primary text-white"
    >
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          115
        </h2>
      </div>
    </div>
    <div
      className="card bg-secondary text-white"
    >
      <div
        className="card-body"
      >
        <h2>
          Vendors: 
          98
        </h2>
      </div>
    </div>
    <div
      className="card bg-info text-white"
    >
      <div
        className="card-header"
      >
        Flavors: 
        8617
      </div>
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          90
        </h2>
      </div>
    </div>
    <div
      className="card bg-warning text-white"
    >
      <div
        className="card-header"
      >
        Recipes: 
        42
      </div>
      <div
        className="card-body"
      >
        <h2>
          Tags: 
          55
        </h2>
      </div>
    </div>
  </div>,
  <a
    href="#home"
    onClick={[Function]}
    title="Refresh Stats"
  >
    <i
      className="fa [object Object]"
    />
  </a>,
]
`;
